applications/MeshingApplication/external_includes/trigen_glass_forming.h

```diff
@@ -0,0 +1,1363 @@
+            double* node2_data = geom[2].SolutionStepData().Data(step);
+
+            //copying this data in the position of the vector we are interested in
+            for(unsigned int j= 0; j<step_data_size; j++)
+            {
+
+                step_data[j] = N[0]*node0_data[j] + N[1]*node1_data[j] + N[2]*node2_data[j];
+
+
+            }
+        }
+        if (N[0]==0.0 && N[1]==0.0 && N[2]==0.0)
+            KRATOS_THROW_ERROR(std::logic_error,"SOMETHING's wrong with the added nodes!!!!!! ERROR","");
+
+        //if ( pnode->FastGetSolutionStepValue(BULK_MODULUS)==0.0)
+        //		KRATOS_THROW_ERROR(std::logic_error,"SOMETHING's wrong with the added nodes!!!!!! ERROR","");
+
+        //now we assure that the flag variables are set coorect!! since we add nodes inside of the fluid volume only
+        //we manually reset the IS_BOUNDARY, IS_FLUID, IS_STRUCTURE, IS_FREE_SURFACE values in a right way
+        //not to have values, like 0.33 0.66 resulting if we would have been interpolating them in the same way
+        //as the normal variables, like Velocity etc
+
+
+        pnode->FastGetSolutionStepValue(IS_BOUNDARY)=0.0;
+        pnode->FastGetSolutionStepValue(IS_STRUCTURE)=0.0;
+        pnode->FastGetSolutionStepValue(IS_INTERFACE)=0.0;
+        pnode->Set(TO_ERASE,false);
+        pnode->FastGetSolutionStepValue(IS_FREE_SURFACE)=0.0;
+        pnode->FastGetSolutionStepValue(IS_FLUID)=1.0;
+    }
+
+	void InterpolateOnEdge( Geometry<Node<3> >& geom, unsigned int step_data_size, Node<3>::Pointer pnode)
+		{
+		unsigned int buffer_size = pnode->GetBufferSize();
+		KRATOS_WATCH(buffer_size)
+
+		for(unsigned int step = 0; step<buffer_size; step++)
+		{	
+				
+			//getting the data of the solution step
+			double* step_data = (pnode)->SolutionStepData().Data(step);		
+			
+											
+			double* node0_data = geom[0].SolutionStepData().Data(step);
+			double* node1_data = geom[1].SolutionStepData().Data(step);						
+				
+			//copying this data in the position of the vector we are interested in
+			for(unsigned int j= 0; j<step_data_size; j++)
+				{ 
+
+					step_data[j] = 0.5*(node0_data[j] + node1_data[j]);
+								
+
+				}						
+			}
+			//now we assure that the flag variables are set coorect!! since we add nodes inside of the fluid volume only
+			//we manually reset the IS_BOUNDARY, IS_FLUID, IS_STRUCTURE, IS_FREE_SURFACE values in a right way
+			//not to have values, like 0.33 0.66 resulting if we would have been interpolating them in the same way 		
+			//as the normal variables, like Velocity etc		
+			
+			//pnode->FastGetSolutionStepValue(IS_BOUNDARY)=1.0;
+			//pnode->FastGetSolutionStepValue(FLAG_VARIABLE)=1.0;			
+
+			pnode->FastGetSolutionStepValue(IS_LAGRANGIAN_INLET)=0.0;
+			pnode->FastGetSolutionStepValue(IS_BOUNDARY)=1.0;
+			pnode->FastGetSolutionStepValue(IS_FREE_SURFACE)=1.0;
+			pnode->FastGetSolutionStepValue(IS_FLUID)=1.0;
+			pnode->FastGetSolutionStepValue(IS_STRUCTURE)=0.0;
```

Comment:
many of these variables can be changed to corresponding flag. (you can leave this for at the moment)